include/exadg/operators/navier_stokes_calculators.h

```diff
@@ -72,6 +109,125 @@ class ShearRateCalculator
   unsigned int quad_index;
 };
 
+template<int dim, typename Number>
+class VorticityCalculator
+{
+public:
+  static unsigned int const number_vorticity_components = (dim == 2) ? 1 : dim;
+
+  typedef dealii::LinearAlgebra::distributed::Vector<Number> VectorType;
+
+  typedef VorticityCalculator<dim, Number> This;
+
+  typedef CellIntegrator<dim, dim, Number> CellIntegratorVector;
+
+  typedef dealii::Tensor<1, dim, dealii::VectorizedArray<Number>> vector;
+
+  VorticityCalculator();
+
+  void
+  initialize(dealii::MatrixFree<dim, Number> const & matrix_free_in,
+             unsigned int const                      dof_index_in,
+             unsigned int const                      quad_index_in);
+
+  void
+  compute_vorticity(VectorType & dst, VectorType const & src) const;
+
+private:
+  void
+  cell_loop(dealii::MatrixFree<dim, Number> const &       matrix_free,
+            VectorType &                                  dst,
+            VectorType const &                            src,
+            std::pair<unsigned int, unsigned int> const & cell_range) const;
+
+  dealii::MatrixFree<dim, Number> const * matrix_free;
+  unsigned int                            dof_index;
+  unsigned int                            quad_index;
+};
+
+template<int dim, typename Number>
+class VelocityMagnitudeCalculator
```

Comment:
We should rename this class to avoid misunderstandings, because it is also used to calculate the vorticity magnitude (if I remember correctly). So I suggest just `MagnitudeCalculator`.

Reply:
Did that, commit is incoming!